### PR TITLE
fix(discover): Serialize Users together to prevent N+1

### PR DESCRIPTION
**Pull request by shruthilayaj**
We were individually serializing Users to get the
User Avatar info, serialize all the user objects
at once so we don't N+1

### CODE REVIEW

File: src/sentry/api/serializers/models/dashboard.py [hunk 1]
```diff
@@ -89,14 +89,27 @@ def get_attrs(self, item_list, user):
 
             result[dashboard]["widget_preview"].append(widget_preview)
 
+        user_serializer = UserSerializer()
+        serialized_users = {
+            user["id"]: user
+            for user in serialize(
+                [dashboard.created_by for dashboard in item_list if dashboard.created_by],
```
Comment (markstory) [inline src/sentry/api/serializers/models/dashboard.py:96]: Should the users be eager loaded with a `select_related`?

File: src/sentry/api/serializers/models/discoversavedquery.py [hunk 2]
```diff
@@ -7,6 +9,30 @@
 
 @register(DiscoverSavedQuery)
 class DiscoverSavedQuerySerializer(Serializer):
+    def get_attrs(self, item_list, user):
+        result = defaultdict(lambda: {"created_by": {}})
+
```
Comment (wedamija) [inline src/sentry/api/serializers/models/discoversavedquery.py:14]: Same comment re `prefetch_related_objects` here for `created_by`

File: src/sentry/api/serializers/models/dashboard.py (hunk 1 above)
Comment (shruthilayaj) [reply src/sentry/api/serializers/models/dashboard.py:96]: We do select related on the users here: https://github.com/getsentry/sentry/blob/master/src/sentry/api/endpoints/organization_dashboards.py#L49
Will look at the prefetch_related_objects

Comment (shruthilayaj) [reply src/sentry/api/serializers/models/dashboard.py:96]: Just got a chance to look at this again, both DiscoverSavedQuery and Dashboards select_related("created_by") in the initial queryset, so the prefetch shouldn't be necessary. I'm not sure if the preferred pattern is to do the select_related in the initial query or the prefetch_related_objects in the serializer. I'll leave it as is for now, I can update if there's a string preference for one/other

Comment (wedamija) [reply src/sentry/api/serializers/models/dashboard.py:96]: I generally recommend doing `prefetch_related_objects ` in the serializers regardless of what outside classes are doing. It won't cause extra work if the related objects have already been fetched. But it means that if someone else uses this serializer in another endpoint they don't have to remember to call `select_related` on it.

**Review by wedamija — APPROVED** [review]

**Review by markstory — APPROVED** [review]

**Pull request merged**